js/main.js

```diff
@@ -76,10 +88,113 @@ function bookmark(event) {
+    if (p2) {
+      div.appendChild(p2);
+    }
+    $ul.appendChild(li);
   } else {
     $bookmarkIcon.setAttribute('src', 'images/bookmark-plus.png');
     data.entries.pop();
+    $ul.removeChild($ul.lastElementChild);
+  }
+
+}
+
+window.addEventListener('DOMContentLoaded', dogList);
+
+function dogList(event) {
+  for (var i = 0; i < data.entries.length; i++) {
+    $ul.appendChild(dogListView(data.entries[i]));
+  }
+}
+
+function dogListView(entry) {
+  var li = document.createElement('li');
+  li.setAttribute('class', 'row justify-end');
+
+  var img = document.createElement('img');
+  img.setAttribute('class', 'dog column-half margin-right-bottom-desktop');
+  img.setAttribute('src', entry[0].url);
+  li.appendChild(img);
+
+  var div = document.createElement('div');
+  div.setAttribute('class', 'column-half');
+  var p = document.createElement('p');
+  div.appendChild(p);
+  if (entry[0].breeds[0]) {
+    p.textContent = entry[0].breeds[0].name;
+    var p2 = document.createElement('p');
+    p2.setAttribute('class', 'inline breed-info margin-none underline margin-bottom text-align-right-mobile width');
+    p2.textContent = 'Click here to learn more!';
+    p2.setAttribute('id', entry[0].id);
+  } else {
+    p.textContent = 'Unknown';
   }
+  p.setAttribute('class', 'inline text-align-right-mobile margin-bottom-none margin-top-none');
+  div.appendChild(p);
+  li.appendChild(div);
+  if (p2) {
+    div.appendChild(p2);
+  }
+  return li;
+}
+
+var iD = null;
+
+function listModal(event) {
+  if (event.target.matches('.breed-info')) {
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i][0].id === event.target.getAttribute('id')) {
+        iD = i;
+      }
+    }
+    $breedName.textContent = data.entries[iD][0].breeds[0].name;
+    $bredFor.textContent = data.entries[iD][0].breeds[0].bred_for;
+    $breedHeight.textContent = data.entries[iD][0].breeds[0].height.imperial + ' inches';
+    $breedWeight.textContent = data.entries[iD][0].breeds[0].weight.imperial + ' lb';
+    $breedTemper.textContent = data.entries[iD][0].breeds[0].temperament;
+    $modal.setAttribute('class', 'modal');
+  }
+}
+
+function showHome(event) {
+  $homePage.setAttribute('class', 'home-page');
+  $viewPage.setAttribute('class', 'view-page hidden');
+  $savedButtonHeader.setAttribute('src', 'images/bookmark-black.png');
+  $homeButtonHeader.setAttribute('src', 'images/house-door.png');
+  $savedButton.setAttribute('src', 'images/bookmark-black.png');
+  $homeButton.setAttribute('src', 'images/house-door.png');
+}
 
+function showSaved(event) {
+  $homePage.setAttribute('class', 'home-page hidden');
+  $viewPage.setAttribute('class', 'view-page');
+  $savedButtonHeader.setAttribute('src', 'images/bookmark-pink.png');
+  $homeButtonHeader.setAttribute('src', 'images/house-black.png');
+  $savedButton.setAttribute('src', 'images/bookmark-pink.png');
+  $homeButton.setAttribute('src', 'images/house-black.png');
 }
```

Comment:
You should make a `switchView` function that can receive an argument of the view to show, just like you did in `code-journal`. That way you don't need to create a new function every time you add a new view.